frappe/model/meta.py

```diff
@@ -363,7 +363,24 @@ def get_dashboard_data(self):
 		data = frappe._dict()
 		try:
 			module = load_doctype_module(self.name, suffix='_dashboard')
-			if hasattr(module, 'get_data'):
+			dashboard_method = False
+
+			for doctype, method in frappe.get_hooks("override_dashboard_data").iteritems():
```

Comment:
why not just `frappe.get_hooks('get_dashboard_data').get(self.name)`

Reply:
I think this will make an exception if theres no key for self.name. I could wrap it in try, except block to solve that. Do you consider this as a better solution? Thank you.

Reply:
You fix it!

Also this will encourage more custom apps, we need contributions. So this better be good.